rendering/cases/layer-clipping/main.js

```diff
@@ -0,0 +1,62 @@
+import Map from '../../../src/ol/Map.js';
+import View from '../../../src/ol/View.js';
+import XYZ from '../../../src/ol/source/XYZ.js';
+import TileLayer from '../../../src/ol/layer/Tile.js';
+import MultiPolygon from '../../../src/ol/geom/MultiPolygon.js';
+import Style from '../../../src/ol/style/Style.js';
+import Stroke from '../../../src/ol/style/Stroke.js';
+
+const source = new XYZ({
+  url: '/data/tiles/osm/{z}/{x}/{y}.png',
+  transition: 0
+});
+
+const layer = new TileLayer({
+  source: source
+});
+
+const geometry = new MultiPolygon([
+  [[[-80, -40], [-40, 0], [-80, 40], [-120, 0], [-80, -40]]],
+  [[[80, -40], [120, 0], [80, 40], [40, 0], [80, -40]]]
+]).transform('EPSG:4326', 'EPSG:3857');
+
+const style = new Style({
+  stroke: new Stroke({
+    width: 2,
+    color: 'blue'
+  })
+});
+
+layer.on('precompose', function(event) {
```

Comment:
There are no 'precompose' and 'postcompose' events any more. Use 'prerender' and 'postrender' instead.